utils/transfer/src/http.rs

```diff
@@ -10,6 +10,36 @@ use futures::{SinkExt, StreamExt, TryStreamExt};
 use std::thread;
 use url::Url;
 
+enum HttpAuth<'s> {
+    None,
+    Basic {
+        username: &'s str,
+        password: Option<&'s str>,
+    },
+}
+
+impl<'s> From<&'s Url> for HttpAuth<'s> {
+    fn from(url: &'s Url) -> Self {
+        if url.username().is_empty() {
+            HttpAuth::None
+        } else {
+            HttpAuth::Basic {
+                username: url.username(),
+                password: url.password(),
+            }
+        }
+    }
+}
+
+fn request(method: Method, url: Url) -> awc::ClientRequest {
+    let builder = awc::ClientBuilder::new();
+    let client = match HttpAuth::from(&url) {
+        HttpAuth::None => builder.finish(),
+        HttpAuth::Basic { username, password } => builder.basic_auth(username, password).finish(),
+    };
```

Comment:
```suggestion
    if let HttpAuth::Basic { username, password } = HttpAuth::from(&url) {
        builder.basic_auth(username, password);
    }
    let client = builder.finish();
```
I think it would be more readable, but I'm not insisting.

Reply:
Unfortunately `basic_auth` consumes and returns `self`, so it cannot be implemented this way. Also, `HttpAuth` will probably soon be extended with new variants